src/ripple/app/ledger/Ledger.cpp

```diff
@@ -245,11 +246,18 @@ void Ledger::updateHash ()
             mAccountHash.zero ();
     }
 
-    // VFALCO TODO Fix this hard coded magic number 122
-    Serializer s (122);
-    s.add32 (HashPrefix::ledgerMaster);
-    addRaw (s);
-    mHash = s.getSHA512Half ();
+    // VFALCO This has to match addRaw
+    mHash = sha512Half(
+        HashPrefix::ledgerMaster,
+        std::uint32_t(mLedgerSeq),
+        std::uint64_t(mTotCoins),
+        mParentHash,
+        mTransHash,
+        mAccountHash,
+        std::uint32_t(mParentCloseTime),
+        std::uint32_t(mCloseTime),
+        std::uint8_t(mCloseResolution),
+        std::uint8_t(mCloseFlags));
```

Comment:
Why are you casting `mLedgerSeq`, `mTotCoins`, 'mParentCloseTime`, and `mCloseTime`? They are declared as the type that you're casting to. (The other two casts make sense, since the member variables are bigger.)